src/fft.rg

```diff
@@ -13,74 +13,135 @@
 -- limitations under the License.
 
 --[[--
-Regent FFT library.
+Regent FFT library
 ]]
 
 import "regent"
 
+local format = require("std/format")
 local data = require("common/data")
-local cudahelper = require("regent/cudahelper")
+local gpuhelper = require("regent/gpu/helper")
+local gpu_available = gpuhelper.check_gpu_available()
 
+-- Import C and FFTW APIs
 local c = regentlib.c
 local fftw_c = terralib.includec("fftw3.h")
-terralib.linklibrary("libfftw3.so")
+regentlib.linklibrary("libfftw3.so")
+regentlib.linklibrary("libfftw3f.so")
 
-local use_cuda = cudahelper.check_cuda_available()
+-- Import cuFFT API
 local cufft_c
-if use_cuda then
-  cufft_c = terralib.includec("cufft.h")
-  terralib.linklibrary("libcufft.so")
+local cufft_assert
+if gpu_available then
+  cufft_c = terralib.includec("cufftXt.h")
+  regentlib.linklibrary("libcufft.so")
+
+  terra cufft_assert(result : cufft_c.cufftResult)
+    var status = "UNKNOWN"
+    if result == 0 then
+      status = "CUFFT_SUCCESS"
+    elseif result == 1 then
+      status = "CUFFT_INVALID_PLAN"
+    elseif result == 2 then
+      status = "CUFFT_ALLOC_FAILED"
+    elseif result == 3 then
+      status = "CUFFT_INVALID_TYPE"
+    elseif result == 4 then
+      status = "CUFFT_INVALID_VALUE"
+    elseif result == 5 then
+      status = "CUFFT_INTERNAL_ERROR"
+    elseif result == 6 then
+      status = "CUFFT_EXEC_FAILED"
+    elseif result == 7 then
+      status = "CUFFT_SETUP_FAILED"
+    elseif result == 8 then
+      status = "CUFFT_INVALID_SIZE"
+    elseif result == 9 then
+      status = "CUFFT_UNALIGNED_DATA"
+    elseif result == 10 then
+      status = "CUFFT_INCOMPLETE_PARAMETER_LIST"
+    elseif result == 11 then
+      status = "CUFFT_INVALID_DEVICE"
+    elseif result == 12 then
+      status = "CUFFT_PARSE_ERROR"
+    elseif result == 13 then
+      status = "CUFFT_NO_WORKSPACE"
+    elseif result == 14 then
+      status = "CUFFT_NOT_IMPLEMENTED"
+    elseif result == 15 then
+      status = "CUFFT_LICENSE_ERROR"
+    elseif result == 16 then
+      status = "CUFFT_NOT_SUPPORTED"
+    end
+    regentlib.assert(result == cufft_c.CUFFT_SUCCESS, status)
+  end
 end
 
--- Hack: get defines from fftw3.h
+-- Define constants: get defines from fftw3.h
```

Comment:
I mean, it is a hack. 😃 

The reason why the "hack" comment is here is because we cannot fetch them automatically (like I asked you to check for cuFFT). We are not "defining" the comments: that's what FFTW does. We're replicating constants that *FFTW defined*. Therefore, while it's a minor point, I'd put the "hack" comment back to reflect what we're really doing here.

Reply:
Added this back